src/traces/isosurface/convert.js

```diff
@@ -154,17 +149,32 @@ function generateIsoMeshes(data) {
     var numVertices;
     var beginVertextLength;
 
-    var Xs = distinctVals(data.x.slice(0, data._len));
-    var Ys = distinctVals(data.y.slice(0, data._len));
-    var Zs = distinctVals(data.z.slice(0, data._len));
+    var Xs = data._Xs;
+    var Ys = data._Ys;
+    var Zs = data._Zs;
 
     var width = Xs.length;
     var height = Ys.length;
     var depth = Zs.length;
 
-    function getIndex(i, j, k) {
-        return k + depth * j + depth * height * i;
-    }
+    var filled = GRID_TYPES.indexOf(data._gridFill.replace(/-/g, '').replace(/\+/g, ''));
```

Comment:
In https://github.com/plotly/plotly.js/issues/4440, you mentioned that `volume` traces are affected by this bug too. So don't you need to bring this new logic over to `volume/convert.js` ?

Reply:
Good eye!

Reply:
Since `volume/convert` reused parts of `isosurface/convert`, no change was needed.

Reply:
Cool! I don't think any of your new mocks have `volume` traces in them. Can you add a new mock to lock down the behaviour for `volume` traces?

Reply:
Good call. Added in https://github.com/plotly/plotly.js/pull/4431/commits/6d062292c01fd69e1c853d653d73a5ed11e9a0f7.
We also have jasmine tests for `volume`.